improver/spotdata/build_spotdata_cube.py

```diff
@@ -135,6 +136,7 @@ def build_spotdata_cube(
         aux_coords_and_dims.append((neighbour_methods_key, neighbour_methods_dim))
 
     if grid_attributes is not None:
+        spot_index_dim += 1
```

Comment:
I don't know if we have a case where this would happen, but if neighbour_methods is None and grid_attributes is not None, both grid_attributes and spot_index will end up trying to be indexed at 1 which will bring about an iris exception.

Reply:
Thanks, I had overlooked this, I've updated the logic so this can't happen now